### PR TITLE
Add plumber support

### PR DESCRIPTION
**Pull request by muguira-james**
This PR includes 2 items:
1) plumber support in the bot
2) a test UI based on ReactJS

The test UI is just scaffolding.  It IS NOT the main UI.  It IS ONLY for testing.  

The bot now provides support for plumber AND cmd line access.  The plumber side requires much more testing and some test cases (TBD)

### CODE REVIEW

File: iPoetMarketResearchBot.r [hunk 1]
```diff
@@ -168,112 +170,76 @@ doInteractive <- function(uniqueLinks) {
...
-	# United Kingdom Navy
-	# subject.matter <- readline(paste("What is the subject matter you are interested in researching? "))
-	subject.matter <- arguments$subject.matter
-	# UK MCM Minewarfare MCM countermeasures USV TRL Speed
-	# seed.words <- readline(paste("What seed words are associated with the subject matter? Seperate words by spaces. "))
-	seed.words <- arguments$seed.words
+#* @filter cors
+cors <- function(res) {
+    res$setHeader("Access-Control-Allow-Origin", "*")
+    plumber::forward()
```
Comment (rossgore) [inline iPoetMarketResearchBot.r]: Is there a reason why we wouldn't just bring in the library at the beginning and then only call "forward()" here? The plumber::forward() syntax feels a little awkward. Nitpicky, I know and not a priority to change but unless there is a good reason I think we should drop the package:: prefix.

File: iPoetMarketResearchBot.r [hunk 2]
```diff
@@ -515,15 +481,91 @@ if ("relevant.urls" %in% names(arguments)) {
 		}
 		cat("\n\n")
 	}
+		# oo <- list(james="hello james", greeting="wow, that's cool")
```
Comment (rossgore) [inline iPoetMarketResearchBot.r]: When you get a chance please clean this out. I'm as guilty as this as anyone but I'd like to keep it clean of testing that is commented out.

File: iPoetMarketResearchBot.r [hunk 3]
```diff
@@ -168,112 +170,76 @@ doInteractive <- function(uniqueLinks) {
...
+#* @param links is a data frame in json format
+#* @param subject_matter is a string
+#* @param seed_words is a string
+#* @param search_term is a string
+#* @post /eval
+handleSearchEvaluation <- function(req, links, search_parameters) {
 	# grab the input file and process the JSON
-	result <- fromJSON(file = arguments$relevant.urls)
 	
+	# result <- fromJSON(relevant_FileName)
```
Comment (rossgore) [inline iPoetMarketResearchBot.r]: I see this is changed from the previous version. When running locally previously I would continually get an error here when trying to parse the JSON I was sending as input. 
[search.txt](https://github.com/rossgore/iPoet/files/2687494/search.txt)

Rscript iPoetMarketResearchBot.r -r search.json 
Error in fromJSON(file = arguments$relevant.urls) : 
  argument "txt" is missing, with no default
Execution halted


I had filled in a few relevant fields in the JSON object generated by the previous function call and nothing else but it always seemed to error out. Hoping this either fixes things or from the sample JSON attached I was using, my user error is very clear. Had to save as txt to upload here, should be json extension.

File: iPoetMarketResearchBot.r [hunk 4]
```diff
@@ -515,15 +481,91 @@ if ("relevant.urls" %in% names(arguments)) {
 		}
 		cat("\n\n")
 	}
+		# oo <- list(james="hello james", greeting="wow, that's cool")
+  	oo <- toJSON(relatedCorpus)
+  	return (oo)
+	
+}
+# =================== main program ===============================
```
Comment (rossgore) [inline iPoetMarketResearchBot.r]: Probably don't need all three sets of comments here.

File: iPoetMarketResearchBot.r [hunk 5]
```diff
@@ -515,15 +481,91 @@ if ("relevant.urls" %in% names(arguments)) {
...
+    make_option(c("-w", "--seed.words"), action="store", help="list of keywords"),
+    make_option(c("-a", "--search.db"), action="store", help="file to store the search parameters"),
+    make_option(c("-r", "--relevant.urls", action="store", help="user suuplied relevant urls")),
+    make_option(c("-b", "--relevant.db"), action="store", help="file to save the scored urls and relevance indicators")
+)
+# takes a file name
+parser <- OptionParser(usage = usage, option_list=option_list)
+arguments <- parse_args(parser)
+
+# opt <- arguments$options
```
Comment (rossgore) [inline iPoetMarketResearchBot.r]: Do we need these two lines?

File: iPoetMarketResearchBot.r (hunk 1 above)
Comment (muguira-james) [reply iPoetMarketResearchBot.r]: I just copied this from stack overflow (fixed now)

File: iPoetMarketResearchBot.r (hunk 2 above)
Comment (muguira-james) [reply iPoetMarketResearchBot.r]: opps, fixed now

File: iPoetMarketResearchBot.r (hunk 3 above)
Comment (muguira-james) [reply iPoetMarketResearchBot.r]: Sorry, the format of this file was a learning experience for me.  I'll attach a sample

jsonlite understood how to turn a json array of objects into a data frame.  The other package wanted list(list(list(...))).

File: iPoetMarketResearchBot.r (hunk 4 above)
Comment (muguira-james) [reply iPoetMarketResearchBot.r]: Ok, I do that so I can find stuff visually.  I keep 1 line

File: iPoetMarketResearchBot.r (hunk 5 above)
Comment (muguira-james) [reply iPoetMarketResearchBot.r]: these lines are important for cmd line processing.

Comment (rossgore) [reply iPoetMarketResearchBot.r]: Got it; should be left then.

**Pull request merged**